ducktape/cluster/remoteaccount.py

```diff
@@ -139,8 +75,29 @@ def __init__(self, ssh_config, externally_routable_ip=None, logger=None):
         self.user = ssh_config.user
         self.externally_routable_ip = externally_routable_ip
         self._logger = logger
-        self._ssh_client = None
-        self._sftp_client = None
+
+    @staticmethod
+    def make_remote_account(ssh_config, externally_routable_ip=None):
+        """Factory function for creating the correct RemoteAccount implementation."""
+
+        # import here to avoid a circular import.
+        from ducktape.cluster.linux_remoteaccount import LinuxRemoteAccount
+        from ducktape.cluster.windows_remoteaccount import WindowsRemoteAccount
+
+        if ssh_config.host and "windows" in ssh_config.host:
+            return WindowsRemoteAccount(ssh_config=ssh_config,
+                                        externally_routable_ip=externally_routable_ip)
+        else:
+            return LinuxRemoteAccount(ssh_config=ssh_config,
+                                      externally_routable_ip=externally_routable_ip)
+
+    def has_operating_system(self, operating_system):
+        # import here to avoid a circular import. TODO: Is there a better way to do this?
```

Comment:
I like your latter suggestion, but it feels to me that the feature isn't needed today and would introduce complexity. Although maybe the complexity feels larger to me because I'm rusty in Python ;). If it's ok with you, I'll get around the circular dependency with your first suggestion and leave the latter suggestion for a future improvement when it's needed.